sdk/Pulumi/Resources/ComponentResource.cs

```diff
@@ -61,7 +62,38 @@ public ComponentResource(string type, string name, ComponentResourceOptions? opt
         /// state as quickly as possible (instead of waiting until the entire application completes).
         /// </summary>
         protected void RegisterOutputs()
-            => RegisterOutputs(ImmutableDictionary<string, object?>.Empty);
+        {
+            var outputs = new Dictionary<string, object?>();
+            var currentType = GetType();
+            foreach (var prop in currentType.GetProperties())
+            {
+                if (prop.Name == nameof(Urn))
+                {
+                    continue;
+                }
+
+                var outputAttribute =
+                    prop.GetCustomAttributes(typeof(OutputAttribute), false)
+                        .FirstOrDefault();
+
+                if (outputAttribute is OutputAttribute attr && attr.Name != null)
```

Comment:
These two blocks are the same _except_ for the name used. Maybe refactor into something like:
```
if is OutputAttribute {
  name = prop.Name
  if attr.Name != null {
    name = attr.Name
  }
  var value = prop.GetValue(this)
  outputs.Add(name, value)
}
```

Reply:
That's nicer! updated the implementation ✅